# <span style="font-size: 20pt; font-family: Montserrat, sans-serif; color: rgb(92, 148, 206); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Graded Assignment 3: 9 to 5</span>

  

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Time to show off your SQL skills! For each question, copy the SQL query you used and make note of the answer.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">The Dataset</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">For this assignment, you will be using the Bureau of Labor Statistics (BLS) Current Employment Survey (CES) results which can be found on</span> [<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(17, 85, 204); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; text-decoration-line: underline; vertical-align: baseline; white-space: pre-wrap;">Kaggle</span>](https://www.kaggle.com/bls/employment)<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Business Issue</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">You are working for the Bureau of Labor Statistics with the United States government and have been approached by your boss with an important meeting request. You have been asked by your supervisor to meet with Dolly Parton whose nonprofit is looking to shed light on the state of employment in the United States. As part of the 9 to 5 project, their research is focused on production and nonsupervisory employees and how those employees fare compared to all employees in the United States. While the data the BLS collects from the CES is publicly available, Dolly Parton and her colleagues need your assistance navigating the thousands of rows in each table in LaborStatisticsDB.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">About the Dataset</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">This dataset comes directly from the Bureau of Labor Statistics’ Current Employment Survey (CES). Here are some things you need to know:</span>

1. <span style="font-size: 10pt; background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">The industry table contains an NAICS code. This is different from the industry code. NAICS stands for North American Industry Classification System.</span>
    
2. <span style="font-size: 10pt; background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Series ID is composed of multiple different codes. CES stands for Current Employment Survey, the name of the survey which collected the data. The industry code as specified by the BLS and the data type code as specified in the datatype table.</span>
    

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Set Up</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">To connect to the database, use the same connection info used during the SQL lessons.&nbsp;</span> 

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">For the assignment, we will be using the LaborStatisticsDB database.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Database Exploration</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">To start with, let’s get to know the database further.</span>

1. <span style="background-color: rgb(255, 255, 255); color: rgb(102, 102, 102); font-family: &quot;Times New Roman&quot;; font-size: 10pt; white-space: pre-wrap;">Use this space to make note of each table in the database, the columns within each table, each column’s data type, and how the tables are connected. You can write this down or draw a diagram. Whatever method helps you get an understanding of what is going on with LaborStatisticsDB.</span>

Diagram of the Database Tables:  [Tables Diagram](https:\github.com\tshililoa\Assignment3_9to5\blob\main\Labor_Statistics_DB_Tables_Diagram.pdf)

In [ ]:
--- File name --Labor_Statistics_DB_Tables_Diagram.pdf

2. <span style="background-color: rgb(255, 255, 255); color: rgb(102, 102, 102); font-family: Arvo, serif; font-size: 10pt; white-space: pre-wrap;">What is the datatype for women employees?</span>

In [1]:
SELECT  distinct  d.data_type_text
FROM  LaborStatisticsDB.dbo.series as s, LaborStatisticsDB.dbo.datatype as d
WHERE d.data_type_code = s.data_type_code
and s.series_title = 'Women employees';


(1 row affected)

Total execution time: 00:00:00.193

data_type_text WOMEN EMPLOYEES

3. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">What is the series id for&nbsp; women employees in the commercial banking industry in the financial activities supersector?</span>

In [3]:
SELECT series_id
FROM LaborStatisticsDB.dbo.series AS s, LaborStatisticsDB.dbo.supersector AS sp , LaborStatisticsDB.dbo.datatype AS d, LaborStatisticsDB.dbo.industry AS i 
WHERE s.supersector_code= sp.supersector_code 
AND s.data_type_code = d.data_type_code
AND i.industry_code = s.industry_code
AND sp.supersector_name LIKE '%Financial activities%'
AND d.data_type_text = 'WOMEN EMPLOYEES'
AND i.industry_name= 'Commercial banking'; 


(2 rows affected)

Total execution time: 00:00:00.118

series_id CES5552211010 CEU5552211010

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Join in on the Fun</span>

<span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Time to start joining! You can choose the type of join you use, just make sure to make a&nbsp; note!</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join&nbsp; annual_2016 with series on series_id. We only want the data in the annual_2016 table to be included in the result.</span>

In [41]:
--using LEFT JOIN
-- only showing top 10 for ease of viewing

SELECT TOP 10 a.*
FROM LaborStatisticsDB.dbo.annual_2016 AS a
LEFT JOIN
LaborStatisticsDB.dbo.series AS s
ON a.series_id = s.series_id;

(10 rows affected)

Total execution time: 00:00:00.331

id series_id year period value footnote_codes original_file 0 CEU5500000007 2016 M13 36.9 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 1 CEU5500000008 2016 M13 26.11 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 2 CEU5500000030 2016 M13 962.73 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 3 CEU5500000031 2016 M13 411.29 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 4 CEU5500000032 2016 M13 11.15 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 5 CEU5500000034 2016 M13 111.6 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 6 CEU5500000035 2016 M13 179.2 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 7 CEU5500000081 2016 M13 236997 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 8 CEU5500000082 2016 M13 6189003 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 9 CEU5552200007 2016 M13 37.3 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join series and datatype on data_type_code</span>

In [40]:
-- using FULL JOIN
-- only showing top 10 for ease of viewing 

SELECT TOP 10*
FROM  LaborStatisticsDB.dbo.series AS s
FULL JOIN 
LaborStatisticsDB.dbo.datatype AS d
ON s.data_type_code = d.data_type_code;

(10 rows affected)

Total execution time: 00:00:00.097

series_id supersector_code industry_code data_type_code seasonal series_title data_type_code data_type_text CES0000000001 0 00000000 1 S All employees 1 ALL EMPLOYEES CES0000000010 0 00000000 10 S Women employees 10 WOMEN EMPLOYEES CES0000000025 0 00000000 25 S All employees 25 ALL EMPLOYEES CES0000000026 0 00000000 26 S All employees 26 ALL EMPLOYEES CES0500000001 5 05000000 1 S All employees 1 ALL EMPLOYEES CES0500000002 5 05000000 2 S Average weekly hours of all employees 2 AVERAGE WEEKLY HOURS OF ALL EMPLOYEES CES0500000003 5 05000000 3 S Average hourly earnings of all employees 3 AVERAGE HOURLY EARNINGS OF ALL EMPLOYEES CES0500000006 5 05000000 6 S Production and nonsupervisory employees 6 PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000007 5 05000000 7 S Average weekly hours of production and nonsupervisory employees 7 AVERAGE WEEKLY HOURS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000008 5 05000000 8 S Average hourly earnings of production and nonsupervisory employees 8 AVERAGE HOURLY EARNINGS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES

3. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join series and industry on industry_code</span>

In [39]:
--Using INNER JOIN 
-- only showing top 10 for ease of reading 
SELECT TOP 10*
FROM  LaborStatisticsDB.dbo.series AS s
INNER JOIN 
LaborStatisticsDB.dbo.industry AS i
ON s.industry_code = i.industry_code;

(10 rows affected)

Total execution time: 00:00:00.157

series_id supersector_code industry_code data_type_code seasonal series_title id industry_code naics_code publishing_status industry_name display_level selectable sort_sequence CES0000000001 0 00000000 1 S All employees 0 0 - B Total nonfarm 0 T 1 CES0000000010 0 00000000 10 S Women employees 0 0 - B Total nonfarm 0 T 1 CES0000000025 0 00000000 25 S All employees 0 0 - B Total nonfarm 0 T 1 CES0000000026 0 00000000 26 S All employees 0 0 - B Total nonfarm 0 T 1 CES0500000001 5 05000000 1 S All employees 1 5000000 - A Total private 1 T 2 CES0500000002 5 05000000 2 S Average weekly hours of all employees 1 5000000 - A Total private 1 T 2 CES0500000003 5 05000000 3 S Average hourly earnings of all employees 1 5000000 - A Total private 1 T 2 CES0500000006 5 05000000 6 S Production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000007 5 05000000 7 S Average weekly hours of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000008 5 05000000 8 S Average hourly earnings of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Aggregate Your Friends and Code some SQL</span>

<span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Put together the following:</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many employees were reported in 2016 in all industries? Round to the nearest whole number.&nbsp; No chopping people into little bits, please.</span>

In [45]:
select FORMAT(ROUND(SUM(an.value), 0, 0), '###,###,###', 'en-us') AS number_of_employees_2016
FROM LaborStatisticsDB.dbo.annual_2016 AS an, LaborStatisticsDB.dbo.series AS s
WHERE an.series_id= s.series_id
AND s.series_title = 'All employees';



(1 row affected)

Total execution time: 00:00:00.104

number_of_employees_2016 2,340,612

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many women employees were reported in 2016 in all industries? Round to the nearest whole number.&nbsp;</span>

In [46]:
select FORMAT(ROUND(SUM(an.value), 0, 0), '###,###,###', 'en-us') AS number_of_women_employees
FROM LaborStatisticsDB.dbo.annual_2016 AS an, LaborStatisticsDB.dbo.series AS s
WHERE an.series_id= s.series_id
ANd s.series_title = 'Women employees';


(1 row affected)

Total execution time: 00:00:00.115

number_of_women_employees 1,125,490

3. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many production/nonsupervisory employees were reported in 2016? Round to the nearest whole number.&nbsp;</span>

In [44]:
select FORMAT(ROUND(SUM(an.value), 0, 0), '###,###,###', 'en-us') AS number_of_employees
FROM LaborStatisticsDB.dbo.annual_2016 AS an, LaborStatisticsDB.dbo.series AS s 
WHERE an.series_id= s.series_id
ANd s.series_title = 'Production and nonsupervisory employees';

(1 row affected)

Total execution time: 00:00:00.099

number_of_employees 1,263,650

4. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In January 2017, what is the average weekly hours worked by production and nonsupervisory employees across all industries?</span>

In [47]:
SELECT avg(jan2017.value) AS average_weekly_hours_worked
FROM LaborStatisticsDB.dbo.january_2017 AS jan2017, LaborStatisticsDB.dbo.series AS s 
WHERE jan2017.series_id= s.series_id
AND s.series_title = 'Average weekly hours of production and nonsupervisory employees';


(1 row affected)

Total execution time: 00:00:00.155

average_weekly_hours_worked 36.05862068965521

5. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">What is the total weekly payroll for production and nonsupervisory employees across all industries in January 2017? Round to the nearest penny, please.</span>

In [70]:
SELECT FORMAT(ROUND(SUM(jan2017.value), 2 ,0), 'C', 'en-us') AS total_weekly_payroll
FROM LaborStatisticsDB.dbo.series AS s, LaborStatisticsDB.dbo.january_2017 AS jan2017
WHERE s.series_id = jan2017.series_id
AND s.series_title = 'Aggregate weekly payrolls of production and nonsupervisory employees';


(1 row affected)

Total execution time: 00:00:00.110

total_weekly_payroll $1,838,753,220.00

6. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In January 2017, which industry was the average weekly hours worked by production and nonsupervisory employees the highest? Which industry was the lowest?</span>

In [58]:
-- industry with highest weekly average hours.

SELECT DISTINCT i.industry_name  
FROM LaborStatisticsDB.dbo.industry AS i, LaborStatisticsDB.dbo.series AS s, LaborStatisticsDB.dbo.january_2017 AS jan2017
WHERE i.industry_code = s.industry_code
AND s.series_id = jan2017.series_id 
AND s.series_title = 'Average weekly hours of production and nonsupervisory employees'
AND jan2017.value = (SELECT MAX(jan_2017.value) 
                    FROM LaborStatisticsDB.dbo.january_2017 AS jan_2017, LaborStatisticsDB.dbo.series AS ss 
                    WHERE jan_2017.series_id= ss.series_id
                    AND ss.series_title = 'Average weekly hours of production and nonsupervisory employees');


(1 row affected)

Total execution time: 00:00:00.105

industry_name Motor vehicle power train components

In [57]:
--Industry with lowest weekly average hours.

SELECT DISTINCT i.industry_name  
FROM LaborStatisticsDB.dbo.industry AS i, LaborStatisticsDB.dbo.series AS s, LaborStatisticsDB.dbo.january_2017 AS jan2017
WHERE i.industry_code = s.industry_code
AND s.series_id = jan2017.series_id 
AND s.series_title = 'Average weekly hours of production and nonsupervisory employees'
AND jan2017.value = (SELECT MIN(jan_2017.value) 
                    FROM LaborStatisticsDB.dbo.january_2017 AS jan_2017, LaborStatisticsDB.dbo.series AS ss 
                    WHERE jan_2017.series_id= ss.series_id
                    AND ss.series_title = 'Average weekly hours of production and nonsupervisory employees');


(1 row affected)

Total execution time: 00:00:00.115

industry_name Fitness and recreational sports centers

7. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In January 2021, which industry was the total weekly payroll for production and nonsupervisory employees the highest? Which industry was the lowest?</span>

In [55]:
-- industry with highest total weekly payroll in January 2017
SELECT TEMP.industry_name FROM
(SELECT TOP 1 MAX(jan2017.value) AS val, i.industry_name 
FROM LaborStatisticsDB.dbo.industry AS i, LaborStatisticsDB.dbo.series AS s, LaborStatisticsDB.dbo.january_2017 AS jan2017
WHERE i.industry_code = s.industry_code
AND s.series_id = jan2017.series_id 
AND s.series_title = 'Aggregate weekly payrolls of production and nonsupervisory employees'
GROUP BY i.industry_name
ORDER BY 1 DESC ) AS TEMP;


(1 row affected)

Total execution time: 00:00:00.135

industry_name Total private

In [31]:
-- industry with lowest total weekly payroll in January 2017

SELECT TEMP.industry_name FROM
(SELECT TOP 1 MIN(jan2017.value) AS val, i.industry_name AS industry_name
FROM LaborStatisticsDB.dbo.industry AS i, LaborStatisticsDB.dbo.series AS s, LaborStatisticsDB.dbo.january_2017 AS jan2017
WHERE i.industry_code = s.industry_code
AND s.series_id = jan2017.series_id
AND s.series_title = 'Aggregate weekly payrolls of production and nonsupervisory employees'
GROUP BY i.industry_name
ORDER BY 1 ASC ) AS TEMP;

(1 row affected)

Total execution time: 00:00:00.193

industry_name Coin-operated laundries and drycleaners

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Subqueries, Unions, Derived Tables, Oh My!</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Write a query that returns the series_id, industry_code, industry_name, and value from the january_2017 table but only if that value is greater than the average value for annual_2016 of data_type_code 82.</span>

In [59]:
SELECT  DISTINCT s.series_id, i.industry_code, i.industry_name, jan2017.value
FROM  LaborStatisticsDB.dbo.series AS s,  LaborStatisticsDB.dbo.industry AS i,  LaborStatisticsDB.dbo.january_2017 AS jan2017
WHERE jan2017.series_id =  s. series_id AND s.industry_code = i.industry_code
AND jan2017.value > (SELECT AVG(an.value) FROM  LaborStatisticsDB.dbo.annual_2016 AS an,  LaborStatisticsDB.dbo.series AS ss 
                       WHERE  an. series_id = ss.series_id 
                       AND ss.data_type_code = 82 );

(377 rows affected)

Total execution time: 00:00:00.239

series_id industry_code industry_name value CES0500000056 5000000 Total private 4239112 CES0500000057 5000000 Total private 110301694 CES0500000081 5000000 Total private 3412651 CES0500000082 5000000 Total private 74498171 CES0600000057 6000000 Goods-producing 21779094 CES0600000082 6000000 Goods-producing 13426820 CES0800000056 8000000 Private service-providing 3432382 CES0800000057 8000000 Private service-providing 88280865 CES0800000081 8000000 Private service-providing 2818821 CES0800000082 8000000 Private service-providing 60914722 CES1000000057 10000000 Mining and logging 966707 CES1021000057 10210000 Mining 927250 CES2000000057 20000000 Construction 7555007 CES2000000082 20000000 Construction 5335274 CES2023600057 20236000 Construction of buildings 1765016 CES2023600082 20236000 Construction of buildings 1008035 CES2023620057 20236200 Nonresidential building 990555 CES2023700057 20237000 Heavy and civil engineering construction 1221764 CES2023700082 20237000 Heavy and civil engineering construction 908374 CES2023800057 20238000 Specialty trade contractors 4595075 CES2023800082 20238000 Specialty trade contractors 3421902 CES2023810057 20238100 Building foundation and exterior contractors 833625 CES2023820057 20238200 Building equipment contractors 2326342 CES2023820082 20238200 Building equipment contractors 1696080 CES2023821057 20238210 Electrical contractors 1039534 CES2023822057 20238220 Plumbing and HVAC contractors 1115270 CES3000000057 30000000 Manufacturing 13282613 CES3000000082 30000000 Manufacturing 7491723 CES3100000057 31000000 Durable goods 8769944 CES3100000082 31000000 Durable goods 4859716 CES3133200057 31332000 Fabricated metal products 1421283 CES3133200082 31332000 Fabricated metal products 871298 CES3133300057 31333000 Machinery 1251217 CES3133400057 31334000 Computer and electronic products 1489440 CES3133600057 31336000 Transportation equipment 2130760 CES3133600082 31336000 Transportation equipment 1257764 CES3133600157 31336001 NULL 1050677 CES3133640057 31336400 Aerospace products and parts 879785 CES3200000057 32000000 Nondurable goods 4496035 CES3200000082 32000000 Nondurable goods 2633628 CES3231100057 32311000 Food manufacturing 1293730 CES3231100082 32311000 Food manufacturing 851406 CES3232500057 32325000 Chemicals 1083956 CES4000000056 40000000 Trade 939889 CES4000000057 40000000 Trade 21232093 CES4000000082 40000000 Trade 14850661 CES4142000057 41420000 Wholesale trade 6858602 CES4142000082 41420000 Wholesale trade 4480134 CES4142300057 41423000 Durable goods 3449034 CES4142300082 41423000 Durable goods 2261006 CES4142340057 41423400 Commercial equipment 915034 CES4142400057 41424000 Nondurable goods 2042100 CES4142400082 41424000 Nondurable goods 1367413 CES4142500057 41425000 Electronic markets and agents and brokers 1359605 CES4142500082 41425000 Electronic markets and agents and brokers 855291 CES4142512057 41425120 Wholesale trade agents and brokers 1317589 CES4142512082 41425120 Wholesale trade agents and brokers 831319 CES4200000057 42000000 Retail trade 8857570 CES4200000082 42000000 Retail trade 6123554 CES4244100057 42441000 Motor vehicle and parts dealers 1740340 CES4244100082 42441000 Motor vehicle and parts dealers 1180297 CES4244110057 42441100 Automobile dealers 1232818 CES4244110082 42441100 Automobile dealers 839595 CES4244111057 42441110 New car dealers 1110063 CES4244500057 42445000 Food and beverage stores 1352705 CES4244500082 42445000 Food and beverage stores 974710 CES4244510057 42445100 Grocery stores 1180650 CES4244510082 42445100 Grocery stores 868430 CES4244511057 42445110 Supermarkets and other grocery stores 1120814 CES4244511082 42445110 Supermarkets and other grocery stores 821282 CES4245200057 42452000 General merchandise stores 1437264 CES4245200082 42452000 General merchandise stores 1065807 CES4300000057 43000000 Transportation and warehousing 4588906 CES4300000082 43000000 Transportation and warehousing 3538683 CES4348400057 43484000 Truck transportat

2. <span style="background-color:rgb(255, 255, 255);font-size:10pt;white-space:pre-wrap;color:rgb(102, 102, 102);font-family:Arvo, serif;">Create a Union table comparing average weekly earnings&nbsp; of production and nonsupervisory employees between annual_16 and january_17 using the data type 30.&nbsp; Round to the nearest penny.&nbsp; You should have a column for the average earnings and a column for the year, and the period.</span>

In [60]:
SELECT ROUND(AVG(value), 2, 0) AS average_earnings, year, period
FROM (SELECT an2016.value, an2016.Year, an2016.period
     FROM  LaborStatisticsDB.dbo.annual_2016 AS an2016, LaborStatisticsDB.dbo.series AS s
     WHERE s.data_type_code = 30
     AND s.series_id = an2016.series_id
     UNION
     SELECT jan2017.value, jan2017.Year, jan2017.period
     FROM  LaborStatisticsDB.dbo.january_2017 AS jan2017,  LaborStatisticsDB.dbo.series AS ss
     WHERE ss.data_type_code = 30
     AND ss.series_id = jan2017.series_id) AS Union_Table
GROUP BY year, period;


(2 rows affected)

Total execution time: 00:00:00.143

average_earnings year period 808.34 2017 M01 797.29 2016 M13

## <span style="font-size:12pt;font-family:Montserrat, sans-serif;color:rgb(53, 74, 95);background-color:rgb(255, 255, 255);font-weight:700;vertical-align:baseline;white-space:pre-wrap;">Summarize Your Results</span>

<span style="font-size:10pt;font-family:Arvo, serif;color:rgb(102, 102, 102);background-color:rgb(255, 255, 255);vertical-align:baseline;white-space:pre-wrap;">With what you know now about the </span> <span style="background-color:rgb(255, 255, 255);color:rgb(102, 102, 102);font-family:&quot;Times New Roman&quot;;font-size:13.3333px;white-space:pre-wrap;">Bureau of Labor Statistics (BLS) Current Employment Survey (CES) results and working with the</span> <span style="color:rgb(102, 102, 102);white-space:pre-wrap;font-family:&quot;Times New Roman&quot;;font-size:13.3333px;">Labor Statistics Database</span><span style="background-color:rgb(255, 255, 255);color:rgb(102, 102, 102);font-family:Arvo, serif;font-size:10pt;white-space:pre-wrap;">, answer the following questions. Note that while this is subjective, you should include relevant data to back up your opinion.</span>

1. <span style="background-color:rgb(255, 255, 255);font-size:10pt;white-space:pre-wrap;color:rgb(102, 102, 102);font-family:Arvo, serif;">During which time period did production and nonsupervisory employees fare better?</span>

With average weekly earnings of 808,34, employees around the January 2017 time period, fared better compared to the employees around the entire year of 2016 (average weekly rate of 797.29)

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In which industries did production and nonsupervisory employees fare better?</span>

 The employees in the Total Private industry fare better because they had the highest total weekly payroll compared to other industries.

3. <span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Now that you have explored the datasets, is there any data or information that you wish you had in this analysis?</span>

I wish I had data that shows how different age groups are comparable, and how earnings are spread out based on race factor.